usr/share/rear/lib/config-functions.sh

```diff
@@ -25,7 +25,9 @@ function SetOSVendorAndVersion () {
             grep -q -i 'debian' /etc/os-release && OS_VENDOR=Debian
             grep -q -i -E '(ubuntu|linuxmint)' /etc/os-release && OS_VENDOR=Ubuntu
             grep -q -i 'arch' /etc/os-release && OS_VENDOR=Arch
-            local version_id=$(grep "^VERSION_ID=" /etc/os-release | cut -d\" -f2 ) # 7
+
+            local version_id
+            version_id="$(. /etc/os-release && echo "$VERSION_ID")"
```

Comment:
Personally I would prefer to use `source` instead of `.`
because explicit wording makes code easier to read
(at least as I perceive it) and it makes it easier
to find code places where some file is sourced.

For example I failed to find where else we use `.`
while it is (relatively) easy to find
where we use `source`.

Reply:
Sure, I'll fix that.

Reply:
I have reworked the parsing completely.  I swear that I tested the changes but apparently I didn't test the final version before making the PR.  Sorry!

The problem is that ReaR declares the `VERSION` variable as `readonly` so the sourcing actually failed and did not process all assignments in `/etc/os-release`.  The reworked code now only processes `VERSION_ID`.

Reply:
Oops!
I also had tested it on my openSUSE Leap 15.5 system
but only directly on command line
where it had "just worked" for me
so I "just approved" the pull request:
```
# ( set -x ; version_id="$(. /etc/os-release && echo "$VERSION_ID")" ; echo "'$version_id'" )
++ . /etc/os-release
+++ NAME='openSUSE Leap'
+++ VERSION=15.5
+++ ID=opensuse-leap
+++ ID_LIKE='suse opensuse'
+++ VERSION_ID=15.5
+++ PRETTY_NAME='openSUSE Leap 15.5'
+++ ANSI_COLOR='0;32'
+++ CPE_NAME=cpe:/o:opensuse:leap:15.5
+++ BUG_REPORT_URL=https://bugs.opensuse.org
+++ HOME_URL=https://www.opensuse.org/
+++ DOCUMENTATION_URL=https://en.opensuse.org/Portal:Leap
+++ LOGO=distributor-logo-Leap
++ echo 15.5
+ version_id=15.5
+ echo ''\''15.5'\'''
'15.5'
```

@lzaoral 
thank you for your explanation
why it does not work from inside ReaR
which reminds me to test things within ReaR
```
# ( set -x ; readonly VERSION=2.7 ; version_id="$(. /etc/os-release && echo "$VERSION_ID")" ; echo "'$version_id'" )
+ readonly VERSION=2.7
+ VERSION=2.7
++ . /etc/os-release
+++ NAME='openSUSE Leap'
+++ VERSION=15.5
bash: VERSION: readonly variable
+ version_id=
+ echo ''\'''\'''
''
```

The unexpected thing is that the whole sourcing a file
fails completely here because I would have falsely assumed
that bash continues sourcing a file when one line fails
just as bash proceeds when a command fails in a script.

Some tests how strange 'source' behaves in certain cases:
```
# cat sourcefile
echo BEGIN
readonly foo=foo
foo=bar
echo END

# ( set -x ; source sourcefile && echo OK || echo FAIL )
+ source sourcefile
++ echo BEGIN
BEGIN
++ readonly foo=foo
++ foo=foo
++ foo=bar
bash: foo: readonly variable

# echo $?
1
```
The readonly variable seems to cause a panic abort
because `&& echo OK || echo FAIL` is ignored.

```
# cat sourcefile
echo BEGIN
if then fi
echo END

# ( set -x ; source sourcefile && echo OK || echo FAIL )
+ source sourcefile
++ echo BEGIN
BEGIN
bash: sourcefile: line 5: syntax error near unexpected token `then'
bash: sourcefile: line 5: `if then fi'
+ echo FAIL
FAIL
```
In contrast a syntax error behaves normal.

```
# cat sourcefile
echo BEGIN
cat qqq
echo END

# ( set -x ; source sourcefile && echo OK || echo FAIL )
+ source sourcefile
++ echo BEGIN
BEGIN
++ cat qqq
cat: qqq: No such file or directory
++ echo END
END
+ echo OK
OK
```
It proceeds after a failed program as expected.

From my point of view the panic abort in case of
a readonly variable looks like a bug in bash.

Reply:
I asked a colleague and he replied
```
If a explicit sub shell (...) is used even dash behaves like this
also the ksh
```

So without the sub shell it works also for me:
```
# cat sourcefile
echo BEGIN
readonly foo=foo
foo=bar
echo END


# set -x ; source sourcefile && echo OK || echo FAIL ; set +x
+ source sourcefile
++ echo BEGIN
BEGIN
++ readonly foo=foo
++ foo=foo
++ foo=bar
bash: foo: readonly variable
++ echo END
END
+ echo OK
OK
+ set +x
```

So lesson learned:
Never 'source' from within a sub shell ;-)